src/transformState.js

```diff
@@ -62,7 +62,25 @@
  * @param {Object[]} transforms
  */
 function transformState(state, transforms) {
-  // write code here
+  for (const transform of transforms) {
+    if (transform.operation === 'addProperties') {
+      Object.assign(state, transform.properties);
+    }
+
+    if (transform.operation === 'removeProperties') {
+      for (const property of transform.properties) {
+        delete state[property];
+      }
+    }
+
+    if (transform.operation === 'clear') {
+      const keys = Object.keys(state);
+
+      for (const key of keys) {
+        delete state[key];
```

Comment:
```suggestion
      for (const key in state) {
        delete state[key];
```

Loop `for...in` iterates through properties of the object. There's no need to create a new array of keys